src/components/cardInfoForm/CardForm/CardForm.tsx

```diff
@@ -0,0 +1,240 @@
+import React, { useState } from 'react';
+import styled from '@emotion/styled';
+import CardInputSection from '../CardInputSection/CardInputSection';
+import CardNumberField from '../CardNumberField/CardNumberField';
+import CardCVCField from '../CardCVCField/CardCVCField';
+import { CARD_IFNO_INPUT_STEP } from '../../../App';
+import CardCompanySelect from '../CardCompanySelect/CardCompanySelect';
+import { CARD_COMPANY_NAME } from '../../constants/cardCompany';
+import CardValidityPeriodField from '../CardValidityPeriodField/CardValidityPeriodField';
+import CardPasswordField from '../CardPasswordField/CardPasswordField';
+import useCardValidationError from '../../../hooks/useCardValidationError';
+import { useNavigate } from 'react-router-dom';
+
+interface CardFormProps {
+  cardNumber: string[];
+  cardCompany: keyof typeof CARD_COMPANY_NAME | undefined;
+  cardCVC: string;
+  cardValidityPeriod: {
+    month: string;
+    year: string;
+  };
+  cardPassword: string;
+  setCardNumber: (cardNumber: string[]) => void;
+  handleChangeCardCompany: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+  setCardValidityPeriod: React.Dispatch<
+    React.SetStateAction<{ month: string; year: string }>
+  >;
+  setCardCVC: (cvc: string) => void;
+  setCardPassword: (password: string) => void;
+}
+
+function CardForm({
+  cardNumber,
+  cardCVC,
+  cardCompany,
+  cardValidityPeriod,
+  cardPassword,
+  setCardNumber,
+  setCardCVC,
+  handleChangeCardCompany,
+  setCardValidityPeriod,
+  setCardPassword,
+}: CardFormProps) {
+  const [show, setShow] = useState({
+    cardCompany: false,
+    cvc: false,
+    password: false,
+  });
+
+  const showNextStep = (step: keyof typeof CARD_IFNO_INPUT_STEP) => {
+    if (show[step]) return;
+
+    setShow((prev) => ({ ...prev, [step]: true }));
+  };
+
+  const {
+    state,
+    validateCardNumber,
+    validateCardCVC,
+    validatePeriod,
+    validatePassword,
+  } = useCardValidationError({
+    cardNumber,
+    cardCVC,
+    cardValidityPeriod,
+    showNextStep,
```

Comment:
네이밍을 이렇게 하면, callback의 실행 시점을 짐작할 수 있는 장점이 있습니다~!

```suggestion
    onChangeStep: showNextStep,
```